### ISSUE TITLE
Enabling SDS requires istio-system namespace 

### ISSUE THREAD

**Issue by Kalimero66** [istio/istio#15929]
**Bug description**
Enabling SDS requires istio-system namespace.
If not installed is istio-system. All NodeAgent are stuck in crash loop because they are not 
 able to find the "istio-security" ConfigMap  even though the configMap is created is the installation  namespace.

- If istio is installed in istio-system. It works perfectly.
- If istio is not installed in istio-system (ex. **istio-afg**) but a ConfigMap named istio-security is create in istio-system it works as well.

It sounds like a istio-system is hardcoded somewhere in the NodeAgent logic.

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
No crash loop for the NodeAgent pods.

**Steps to reproduce the bug**
~~~
helm template install/kubernetes/helm/istio --name istio-sds-demo --namespace **istio-afg** --values install/kubernetes/helm/istio/values-istio-demo.yaml > istio-sds-demo.yaml
~~~
~~~
cat values install/kubernetes/helm/istio/values-istio-demo.yaml
~~~
~~~
# This is used to generate minimal demo mode. It is included from demo and demo-auth values.
# It is shipped with the release, used for bookinfo or quick installation of istio.
# Includes components used in the demo, defaults to alpha3 rules.
# Note: please only put common configuration for the demo profiles here.
global:
  proxy:
    accessLogFile: "/dev/stdout"
    resources:
      requests:
        cpu: 10m
        memory: 40Mi

  disablePolicyChecks: false

  controlPlaneSecurityEnabled: false

  mtls:
    # Default setting for service-to-service mtls. Can be set explicitly using
    # destination rules or service annotations.
    enabled: false
  sds:
    enabled: true
    udsPath: "unix:/var/run/sds/uds_path"
    useNormalJwt: true  

sidecarInjectorWebhook:
  enabled: true
  # If true, webhook or istioctl injector will rewrite PodSpec for liveness
  # health check to redirect request to sidecar. This makes liveness check work
  # even when mTLS is enabled.
  rewriteAppHTTPProbe: false

pilot:
  traceSampling: 100.0
  resources:
    requests:
      cpu: 10m
      memory: 100Mi

mixer:
  policy:
    enabled: true
    resources:
      requests:
        cpu: 10m
        memory: 100Mi

  telemetry:
    enabled: true
    resources:
      requests:
        cpu: 50m
        memory: 100Mi
 
  adapters:
    stdio:
      enabled: true
 
grafana:
  enabled: true

tracing:
  enabled: true

kiali:
  enabled: true
  createDemoSecret: true
  dashboard:
    jaegerURL: "http://tracing.rnd/jaeger/search"
    grafanaURL: "http://grafana.rnd"  

gateways:
  istio-ingressgateway:
    resources:
      requests:
        cpu: 10m
        memory: 40Mi

  istio-egressgateway:
    enabled: true
    resources:
      requests:
        cpu: 10m
        memory: 40Mi

nodeagent:
  enabled: true
  image: node-agent-k8s
  env:
    CA_PROVIDER: "Citadel"
    CA_ADDR: "istio-citadel:8060"
    VALID_TOKEN: true

certmanager:
  enabled: true
  email: xxx@xn.com
~~~

**Version (include the output of `istioctl version --remote` and `kubectl version`)**
~~~
-> k version 
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.1", GitCommit:"4485c6f18cee9a5d3c3b4e523bd27972b1b53892", GitTreeState:"clean", BuildDate:"2019-07-18T09:18:22Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-04-03T04:12:37Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
~~~

**How was Istio installed?**
Helm template. See value.yaml above.

**Environment where bug was observed (cloud vendor, OS, etc)**
On prem. 3 masters and 16 nodes.

NodeAgent pods can not find the istio-security configmap and get stuck in crash loop.

~~~
> kubectl get po -n istio-afg
NAME                                      READY   STATUS             RESTARTS   AGE
certmanager-cf775586-t62s7                1/1     Running            0          10m
grafana-5bfd8bcf79-qxkjs                  1/1     Running            0          10m
istio-citadel-dc7fcc945-p7d2p             1/1     Running            0          10m
istio-cleanup-secrets-1.2.2-fczbc         0/1     Completed          0          11m
istio-egressgateway-7cdc657f9c-qvjfx      1/1     Running            0          10m
istio-galley-6f444dc468-z9qxv             1/1     Running            0          10m
istio-grafana-post-install-1.2.2-lr9dz    0/1     Completed          0          11m
istio-ingressgateway-649d564d49-qgxns     1/1     Running            0          10m
istio-nodeagent-25dzs                     0/1     CrashLoopBackOff   2          10m
istio-nodeagent-5vkxz                     0/1     CrashLoopBackOff   2          10m
istio-nodeagent-7nsh2                     1/1     Running            3          10m
istio-nodeagent-8dslp                     0/1     CrashLoopBackOff   2          10m
istio-nodeagent-9wljx                     1/1     Running            3          10m
istio-nodeagent-bnvz2                     1/1     Running            3          10m
istio-nodeagent-bvxjx                     0/1     CrashLoopBackOff   2          10m
istio-nodeagent-d9kc9                     1/1     Running            3          10m
istio-nodeagent-njktl                     1/1     Running            3          10m
istio-nodeagent-nx7s7                     1/1     Running            3          10m
istio-nodeagent-rzs7k                     1/1     Running            3          10m
istio-nodeagent-tclsn                     1/1     Running            3          10m
istio-nodeagent-wpr7v                     0/1     CrashLoopBackOff   2          10m
istio-nodeagent-zh5vd                     0/1     CrashLoopBackOff   2          10m
istio-nodeagent-zm6zc                     0/1     CrashLoopBackOff   2          10m
istio-nodeagent-znzcs                     1/1     Running            3          10m
istio-pilot-68bdb7886-vjkcb               2/2     Running            0          10m
istio-policy-75865d8b75-trckw             2/2     Running            2          10m
istio-security-post-install-1.2.2-jv9rk   0/1     Completed          0          11m
istio-sidecar-injector-597d555984-z52f4   1/1     Running            0          10m
istio-telemetry-7bc4685856-m57vr          2/2     Running            2          10m
istio-tracing-6bfb484c49-xqhq4            1/1     Running            0          10m
kiali-7d6f496c94-6g7r5                    1/1     Running            0          10m
prometheus-5497d6bf6f-kmxk5               1/1     Running            0          10m
~~~

~~~
kubectl logs  istio-nodeagent-7nsh2 -f 


2019-07-30T16:57:03.778086Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:05.782959Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:07.786851Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:09.791172Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:11.796389Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:13.799852Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:15.803612Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:17.806967Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:19.810673Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:21.813957Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:23.817783Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:25.821644Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:27.824727Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:29.831972Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:31.835198Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:33.844332Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:35.846855Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:37.850075Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:39.853243Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:41.856574Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:43.867838Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:45.871356Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:47.875691Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:49.879444Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:51.882765Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:53.892670Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:55.896802Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:57.901179Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:57:59.904384Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:58:01.908934Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:58:03.925458Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:58:05.929965Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:58:07.933467Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:58:09.936732Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
2019-07-30T16:58:11.940019Z	info	unalbe to fetch CA TLS root cert: failed to get CA TLS root cert: configmaps "istio-security" not found, retry in 2s
~~~

**Comment by howardjohn** [comment]
@quanjielin I thought I remmebr you fixing this, is it just not released yet?

**Comment by quanjielin** [comment]
the fix was https://github.com/istio/istio/pull/15199
@Kalimero66 which build you used ?

**Comment by Kalimero66** [comment]
@quanjielin Istio1.2.2
Note that installation in a different namespace works well. Only the NodeAgent is having this issue and this is prevent from using SDS in a non istio-system namespace.

**Comment by quanjielin** [comment]
@Kalimero66 the fix was not in 1.2.2, it's only checked in to master branch.
let me cherrypick it.

**Comment by Kalimero66** [comment]
Thanku.
